src/app/admin/websiteDataTable/WebsiteDataTable.tsx

```diff
@@ -0,0 +1,231 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
+import {
+    DataGrid,
+    GridActionsCellItem,
+    GridColDef,
+    GridEventListener,
+    GridRowEditStopReasons,
+    GridRowId,
+    GridRowModes,
+    GridRowModesModel,
+    useGridApiRef,
+} from "@mui/x-data-grid";
+import EditIcon from "@mui/icons-material/Edit";
+import SaveIcon from "@mui/icons-material/Save";
+import CancelIcon from "@mui/icons-material/Close";
+import { LinearProgress } from "@mui/material";
+import { DatabaseError } from "@/app/errorClasses";
+import { fetchWebsiteData, updateDbWebsiteData } from "./FetchWebsiteData";
+import CustomComponent from "./CustomEditComponent";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+
+export interface WebsiteDataRow {
+    readableName: string;
+    id: string;
+    value: string;
+}
+
+const WebsiteDataTable: React.FC = () => {
+    const [rows, setRows] = useState<WebsiteDataRow[]>([]);
+    const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const apiRef = useGridApiRef();
+
+    useEffect(() => {
+        fetchWebsiteData()
+            .then((response) => setRows(response))
+            .catch(async (error) => {
+                const logId = await logErrorReturnLogId("Error with fetch: website data", error);
+                throw new DatabaseError("fetch", "website data table", logId);
+            })
+            .finally(() => setIsLoading(false));
```

Comment:
do we also want to set isloading to true just before fetching?

Reply:
good catch